internal/controllers/apis/configmap/disable.go

```diff
@@ -0,0 +1,65 @@
+package configmap
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+)
+
+func disableStatus(ctx context.Context, client client.Client, disableSource *v1alpha1.DisableSource) (isDisabled bool, reason string, err error) {
+	if disableSource == nil {
+		return false, "object does not specify a DisableSource", nil
+	}
+
+	if disableSource.ConfigMap == nil {
+		return false, "DisableSource specifies no ConfigMap", nil
+	}
+
+	cm := &v1alpha1.ConfigMap{}
+	err = client.Get(ctx, types.NamespacedName{Name: disableSource.ConfigMap.Name}, cm)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, fmt.Sprintf("ConfigMap %q does not exist", disableSource.ConfigMap.Name), nil
+		}
+		return false, fmt.Sprintf("error reading ConfigMap %q", disableSource.ConfigMap.Name), err
+	}
+
+	cmVal, ok := cm.Data[disableSource.ConfigMap.Key]
+	if !ok {
+		return false, fmt.Sprintf("ConfigMap %q has no key %q", disableSource.ConfigMap.Name, disableSource.ConfigMap.Key), nil
+	}
+	// TODO(matt)
+	// checking `== "true"` rather than strconv.ParseBool because we lack a good way to surface errors
```

Comment:
Hmmm...I'm having trouble reconciling the statement "we lack a good way to surface errors" and then....most of this function is surfacing errors in the reason field? what am i missing?

seems like we could use strconv and then surface any errors in 'reason'

Reply:
Ah, yeah, this code is confusing and silly.

Thanks!